src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -385,4 +394,168 @@ await repo.ExecuteGitCommand(
+
+        Queue<string?> directoriesToProcess = [];
+        directoriesToProcess.Enqueue(null);
+
+        Dictionary<string, string> fileDifferences = [];
+        string vmrMappingPath = $"{VmrInfo.SourcesDir}/{sourceMapping}";
+        var sourceManifest = SourceManifest.FromJson(
+            await vmrGitClient.GetFileContentsAsync(VmrInfo.DefaultRelativeSourceManifestPath, vmrRepo.Remote, vmrRepo.Ref));
+
+        string? currentPath;
+        while (directoriesToProcess.Count > 0)
+        {
+            currentPath = directoriesToProcess.Dequeue();
+
+            var repoFiles = await sourceGitClient.LsTree(sourceRepo.Remote, sourceRepo.Ref, currentPath);
+            var vmrFiles = (await vmrGitClient.LsTree(vmrRepo.Remote, vmrRepo.Ref, $"{vmrMappingPath}{currentPath}"))
+                .Select(item => item with { Path = item.Path.Substring(vmrMappingPath.Length) })
+                .ToList();
+
+            // Blobs with the same content have the same sha, so we need to take that into consideration
+            var filesOnlyInVmr = vmrFiles
+                .GroupBy(f => f.Sha)
+                .ToDictionary(group => group.Key, group => group.ToList());
+
+            foreach (var sourceFile in repoFiles)
+            {
+                if (TryFindFileInVmrAndUpdateFilesOnlyInVmr(sourceFile, filesOnlyInVmr))
+                {
+                    continue;
+                }
+
+                if (sourceFile.IsCommit())
+                {
+                    HandleSubmodule(sourceFile, sourceManifest, fileDifferences, filesOnlyInVmr);
+                }
+                else if (sourceFile.IsBlob())
+                {
+                    RecordBlobDiff(sourceFile, vmrFiles, fileDifferences);
+                }
+                else if (sourceFile.IsTree())
+                {
+                    if (vmrFiles.Any(vmr => vmr.Path == sourceFile.Path))
+                    {
+                        // the folder exists, but the contents of it changed
+                        directoriesToProcess.Enqueue(sourceFile.Path);
+                    }
+                    else
+                    {
+                        fileDifferences[sourceFile.Path] = ($"- tree {sourceFile.Path}");
+                    }
+                }
+            }
+
+            ProcessVmrOnlyFiles(filesOnlyInVmr, fileDifferences, directoriesToProcess);
+        }
+
+        if (fileDifferences.Count == 0)
+        {
+            Console.WriteLine("No differences found between the product repo and the VMR.");
+            return;
+        }
+
+        Console.WriteLine("Differences found between the product repo and the VMR");
+        Console.WriteLine("* means the file is different in the source repo and in the VMR");
+        Console.WriteLine("+ means the file exists in the VMR and not in the source repo");
+        Console.WriteLine("- means the file exists in the source repo and not in the VMR");
+        Console.WriteLine();
```

Comment:
Let's just output a list of files without English text around. This is hard to parse with other scripts or tools and not interesting. If this is required to show somewhere, put it in the command help string.

Or log it with debug level so that `--debug` shows this.

No diff should return empty stdout.